server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2521,6 +2521,13 @@ public UserVm updateVirtualMachine(UpdateVMCmd cmd) throws ResourceUnavailableEx
         String extraConfig = cmd.getExtraConfig();
 
         UserVmVO vmInstance = _vmDao.findById(cmd.getId());
+        if (details.size() > 0) {
```

Comment:
Maybe `MapUtils.isNotEmpty(details)` to prevent possible NPE?

Reply:
Sure